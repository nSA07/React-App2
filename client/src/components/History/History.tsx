import { Added } from './components/Added/Added';
import { Title } from './components/Title/Title';
import { Description } from './components/Description/Description';
import { Priority } from './components/Priority/Priority';
import { List } from './components/List/List';
import { Remove } from './components/Remove/Remove';



export const History = ({type, ...restProps}) => {
    let Component = null

    switch (type) {
        case 'added':
            Component = Added
            break;

            case 'title':
            Component = Title
            break;

            case 'description':
            Component = Description
            break;

            case 'priority':
            Component = Priority
            break;

            case 'list':
            Component = List
            break;

            case 'remove':
            Component = Remove
            break;
    
        default:
            break;
    }

    return <Component {...restProps} />
}