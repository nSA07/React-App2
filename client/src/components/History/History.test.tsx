import { render } from '@testing-library/react'
import { History } from './History';

jest.mock('./components/Added/Added', () => ({
    Added: () => <h1>Added</h1>
}));

jest.mock('./components/Title/Title', () => ({
    Title: () => <h1>Title</h1>
}));

jest.mock('./components/Description/Description', () => ({
    Description: () => <h1>Description</h1>
}));

jest.mock('./components/Priority/Priority', () => ({
    Priority: () => <h1>Priority</h1>
}));

jest.mock('./components/List/List', () => ({
    List: () => <h1>List</h1>
}));

jest.mock('./components/Remove/Remove', () => ({
    Remove: () => <h1>Remove</h1>
}));

describe('<History />', () => {

    test('History component should render Added Component', () => {
        const {container, getAllByText} = render(<History type="added" />)

        expect(getAllByText('Added')).toBeDefined();
        expect(container.firstChild).toMatchSnapshot();
    })

    test('History component should render Title Component', () => {
        const {container, getAllByText} = render(<History type="title" />)

        expect(getAllByText('Title')).toBeDefined();
        expect(container.firstChild).toMatchSnapshot();
    })

    test('History component should render Priority Component', () => {
        const {container, getAllByText} = render(<History type="priority" />)

        expect(getAllByText('Priority')).toBeDefined();
        expect(container.firstChild).toMatchSnapshot();
    })

    test('History component should render List Component', () => {
        const {container, getAllByText} = render(<History type="list" />)

        expect(getAllByText('List')).toBeDefined();
        expect(container.firstChild).toMatchSnapshot();
    })

    test('History component should render Remove Component', () => {
        const {container, getAllByText} = render(<History type="remove" />)

        expect(getAllByText('Remove')).toBeDefined();
        expect(container.firstChild).toMatchSnapshot();
    })
})
