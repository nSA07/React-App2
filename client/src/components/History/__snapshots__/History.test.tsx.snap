// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<History /> History component should render Added Component 1`] = `
<h1>
  Added
</h1>
`;

exports[`<History /> History component should render Added Component 2`] = `
<h1>
  Title
</h1>
`;

exports[`<History /> History component should render Added Component 3`] = `
<h1>
  Priority
</h1>
`;

exports[`<History /> History component should render Added Component 4`] = `
<h1>
  List
</h1>
`;

exports[`<History /> History component should render Added Component 5`] = `
<h1>
  Remove
</h1>
`;

exports[`<History /> History component should render List Component 1`] = `
<h1>
  List
</h1>
`;

exports[`<History /> History component should render Priority Component 1`] = `
<h1>
  Priority
</h1>
`;

exports[`<History /> History component should render Remove Component 1`] = `
<h1>
  Remove
</h1>
`;

exports[`<History /> History component should render Title Component 1`] = `
<h1>
  Title
</h1>
`;
